src/Cli/dotnet/commands/dotnet-msbuild/MSBuildLogger.cs

```diff
@@ -93,23 +87,7 @@ internal static void FormatAndSend(ITelemetry telemetry, TelemetryEventArgs args
             if (args.EventName == TargetFrameworkTelemetryEventName)
             {
                 var newEventName = $"msbuild/{TargetFrameworkTelemetryEventName}";
-                Dictionary<string, string> maskedProperties = new Dictionary<string, string>();
-
-                foreach (var key in new[] {
-                    TargetFrameworkVersionTelemetryPropertyKey,
-                    RuntimeIdentifierTelemetryPropertyKey,
-                    SelfContainedTelemetryPropertyKey,
-                    UseApphostTelemetryPropertyKey,
-                    OutputTypeTelemetryPropertyKey
-                })
```

Comment:
This is the filter we missed before - having this meant we both needed to collect data points declaratively in MSBuild code, but then also sync the fields back here. This seems fragile, so I think we should move to the built-in hashing I added a bit ago.

Reply:
Devil's Advocate - @dsplaisted wondered if there was a potential for new data points to 'leak' out unexpectedly without this filter. I think this risk is fairly small, as we are already constraining the properties in MSBuild props, but we should consider this in more detail.

Reply:
We mitigated this risk by defaulting all collected data points to hashing by default. Individual data points must set the `Hash=false` metadata to disable this behavior.